internal/handler/impl/domain_handler.go

```diff
@@ -330,6 +330,15 @@ func (a *application) UpdateDomainAgent(ctx echo.Context, domain_id uuid.UUID, p
 		)
 	}
 
+	if data.IpaDomain != nil && currentData.IpaDomain != nil &&
+		data.IpaDomain.RealmName != nil && currentData.IpaDomain.RealmName != nil &&
+		*data.IpaDomain.RealmName != *currentData.IpaDomain.RealmName {
+		return internal_errors.NewHTTPErrorF(
+			http.StatusBadRequest,
+			"'realm_name' may not be changed",
+		)
+	}
```

Comment:
optional: I suggest to extract in a function to reduce ciclomatic complexity indicator for the method. BTW the code is working as it is.

```suggestion
       // Check the realm does not change
       if err = a.updateDomainAgentCheckRealmName(data, currentData); err != nil {
               return err
       }
```

And the function just encapsulate the `if` statement at `domain_handler_private.go` file:

```go
func (a *application) updateDomainAgentCheckRealmName(data, currentData *model.Domain) error {
       if data.IpaDomain != nil && currentData.IpaDomain != nil &&
               data.IpaDomain.RealmName != nil && currentData.IpaDomain.RealmName != nil &&
               *data.IpaDomain.RealmName != *currentData.IpaDomain.RealmName {
               return internal_errors.NewHTTPErrorF(
                       http.StatusBadRequest,
                       "'realm_name' may not be changed",
               )
       }
       return nil
}
```

Reply:
I'll skip it as, too keep consistency, it would need to be done also in the block above that checks the `title`.  

+ it is end of day before my PTO so we don't have time for another review round. 

btw, if I'd do it, I'd name the method e.g., `realmNameNotChangedCheck` or `realmNameNotChangedGuard` - to be more self-explanatory.